src/lib/components/Access/utils.js

```diff
@@ -13,44 +13,93 @@ import {
   TextField,
 } from 'react-invenio-forms';
 
-import { Embargo } from './Embargo';
-
-/**
- * Returns the props for a protection button.
- * @param active is button active
- * @param activeColor button color when active
- */
-export function getProtectionButtonProps(active, activeColor) {
-  let props = {active};
-  if (active) {
-    props["color"] = activeColor;
+import { Embargo, EmbargoState } from './Embargo';
+import { FastField, Field } from 'formik';
+
+
+class ProtectionButtonsComponent extends Component {
+  /**
+   * Returns the props for a protection button.
+   * @param active is button active
+   * @param activeColor button color when active
+   */
+  getButtonProps(active, activeColor) {
+    let props = {active};
+    if (active) {
+      props["color"] = activeColor;
+    }
+    return props;
+  }
+
+  render() {
+    const {
+      formik,
+      active,
+    } = this.props;
+
+    return (
+      <Button.Group widths={'2'}>
+        <Button
+          {...this.getButtonProps(active, "green")}
+          onClick={(event, data) => {
+            formik.form.setFieldValue(formik.field.name, "public");
+            // NOTE: We reset values, so if embargo filled and click Public,
+            //       user needs to fill embargo again. Otherwise lots of
+            //       bookkeeping.
+            formik.form.setFieldValue("access.embargo", {
+              active: false
+            });
+          }}
+          compact
+          attached
+        >
+          Public
+        </Button>
+        <Button
+          {...this.getButtonProps(!active, "red")}
+          onClick={(event, data) => formik.form.setFieldValue(formik.field.name, "restricted")}
```

Comment:
minor: Just to follow the same patterns as in the rest of the codebase maybe we can use here the passed prop `fieldPath`.

Reply:
Sure. I do find relying on `formik.field.name` to be slightly cleaner overall in that it reduces the number of passed props.

Reply:
I am ok with it if we normalize it everywhere even I am afraid that might not be feasible everywhere.....I think is worth logging an issue to reiterate through the components and see if this can be applied, especially if it cleans up the props needed!